src/poliastro/atmosphere/models.py

```diff
@@ -0,0 +1,661 @@
+    |  120.0 | 117.777 |  360.00 | 2.5382e-5 |    2.222e-8   |  exponential  |
+    +--------+---------+---------+-----------+---------------+---------------+
+    |  500.0 | 463.540 |  999.24 | 3.0236e-9 |   5.215e-13   |  exponential  |
+    +--------+---------+---------+-----------+---------------+---------------+
+    | 1000.0 | 864.071 |   1000  | 7.5138e-5 |   3.561e-15   |  exponential  |
+    +--------+---------+---------+-----------+---------------+---------------+
+
+    """
+
+    def __init__(self):
+
+        # Lower layers (Z < 86km)
+        Hb_lower = [0.0, 11.0, 20.0, 32.0, 47.0, 51.0, 71.0] * u.km
+        Zb_lower = [
+            geopotential_to_geometric(H, R_earth).to(u.km).value for H in Hb_lower
+        ] * u.km
+
+        # Upper layers (Z >= 86km)
+        Zb_upper = [86.0, 91.0, 110.0, 120.0, 500.0, 1000.0] * u.km
+        Hb_upper = [
+            geometric_to_geopotential(Z, R_earth).to(u.km).value for Z in Zb_upper
+        ] * u.km
+
+        # Geopotential and geometric layers
+        Hb_table = np.append(Hb_lower, Hb_upper).value * u.km
+        Zb_table = np.append(Zb_lower, Zb_upper).value * u.km
+
+        # Temperature layer
+        Tb_table = [
+            288.15,
+            216.65,
+            216.650,
+            228.65,
+            270.65,
+            270.65,
+            214.65,
+            186.95,
+            187.36,
+            254.93,
+            397.91,
+            2019.69,
+            7351.15,
+        ] * u.K
+
+        # Temperature gradient
+        Lb_table = (
+            [-6.5, 0.0, 1.0, 2.8, 0.0, -2.8, -2.0, 0.0, 0.0, 12.0, 0.0, 0.0, 0.0]
+            * u.K
+            / u.km
+        )
+
+        # Pressure layer
+        pb_table = [
+            1.01325e3,
+            2.2632e2,
+            5.4748e1,
+            8.6801e0,
+            1.1090e0,
+            6.6938e-1,
+            3.9564e-2,
+            3.7338e-3,
+            1.5381e-3,
+            7.1042e-5,
+            2.5382e-5,
+            3.0236e-9,
+            7.5138e-11,
+        ] * u.mbar
+
+        # Generate table
+        table_coesa76 = [Zb_table, Hb_table, Tb_table, Lb_table, pb_table]
+        COESA.__init__(self, table_coesa76)
+
+    def temperature(self, alt, geometric=True):
+        """ Returns temperature at given height.
+
+        Parameters
+        ----------
+        alt: ~astropy.units.Quantity
+            Geometric/Geopotential height.
+        geometric: boolean
+            If `True`, assumes that `alt` argument is geometric kind.
+
+        Returns
+        -------
+        T: ~astropy.units.Quantity
+            Temperature at given height.
+        """
+        # Check if valid range and convert to geopotential
+        Z, H = self._get_heights(alt, geometric=geometric)
+        Zb, Hb, Tb, Lb, _ = self._get_base_parameters(H)
+
+        # Different altitudes imply different equations
+        # TODO: Include mean air molecular weight corrections
```

Comment:
Does this `TODO` affect all the layers, or only the last one?

Reply:
No, it only affects the layers between 80km and 86km. Below 80km the mean molecular weight of the air can be assumed to be constant but above this value, till the 86km layer, a small correction may be applied.

At 80km the ratio M/M0 is known to be 1.0, while at 86km this ratio becomes 0.999579, being M the real molecular weight and M0 the one at sea.

I will rewrite this #TODO so it becomes clear what is required :+1: